projects/aca-content/src/lib/components/common/logout/logout.component.ts

```diff
@@ -35,10 +36,37 @@ import { AppStore, SetSelectedNodesAction } from '@alfresco/aca-shared/store';
     </button>
   `
 })
-export class LogoutComponent {
-  constructor(private store: Store<AppStore>) {}
+export class LogoutComponent implements OnInit {
+  public readonly LAST_VISITED_LOCATION_KEY = 'lastVisitedLocation';
+  constructor(private store: Store<AppStore>, private router: Router) {}
+
+  ngOnInit() {
+    this.navigateToLastVisitedLocation();
+  }
 
   onLogoutEvent() {
     this.store.dispatch(new SetSelectedNodesAction([]));
+    const lastVisitedLocation = this.router.url;
+    this.saveLastVisitedLocation(lastVisitedLocation);
+  }
+
+  navigateToLastVisitedLocation() {
+    const lastVisitedLocation = this.getLastVisitedLocation();
+    if (lastVisitedLocation) {
+      this.router.navigateByUrl(lastVisitedLocation);
+    } else {
+      this.navigateToLandingPage();
+    }
+  }
+
+  navigateToLandingPage() {
+    this.router.navigateByUrl('/personal-files');
+  }
+
+  saveLastVisitedLocation(location: string) {
+    localStorage.setItem(this.LAST_VISITED_LOCATION_KEY, location);
```

Comment:
this is not going to work correctly when switching between different user accounts with different permissions to the content, and user home folders